### PR TITLE
Make pass list optional in config

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Make pass list optional in config. If no pass provided, Olive will only evaluate input model.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/engine/engine.py [hunk 1]
```diff
@@ -309,7 +307,6 @@ def run(
                 1. Final model -> {output_dir}/{output_name}_{AcceleratorSpec}_model.onnx
                 2. JSON file -> {output_dir}/{output_name}_{AcceleratorSpec}_model.json
                 3. Evaluation results of the final model -> {output_dir}/{output_name}_{AcceleratorSpec}_metrics.json
-
```
Comment (guotuofeng) [inline olive/engine/engine.py:312]: I add the blank lines to fix the warning when building docs

**Comment by guotuofeng** [comment]
Do we really need only evaluate the input models in Olive?

File: olive/engine/engine.py (hunk 1 above)
Comment (xiaoyu-work) [reply olive/engine/engine.py:312]: I didn't realize this change. I'll add it back.

**Comment by xiaoyu-work** [comment]
> Do we really need only evaluate the input models in Olive?

I will add model perf compare pipeline which will only evaluates input model with no pass added. Actually we added input model only evaluation before but not complete this: https://github.com/microsoft/Olive/blob/main/olive/engine/engine.py#L350

File: olive/engine/engine.py [hunk 2]
```diff
@@ -294,7 +293,6 @@ def run(
     ):
         """
         Run all the registered Olive passes on the input model and produce one or more candidate models.
-
```
Comment (guotuofeng) [inline olive/engine/engine.py:297]: ditto

Comment (guotuofeng) [reply olive/engine/engine.py:297]: maybe you can rebase from main?

Comment (xiaoyu-work) [reply olive/engine/engine.py]: fixed

File: olive/workflows/run/config.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ class RunConfig(ConfigBase):
     evaluators: Dict[str, OliveEvaluatorConfig] = None
     pass_flows: List[List[str]] = None
     engine: RunEngineConfig
-    passes: Dict[str, RunPassConfig]
+    passes: Dict[str, RunPassConfig] = None
```
Comment (guotuofeng) [inline olive/workflows/run/config.py:69]: just curious if the empty dict {} works for original code without your changes. 
This is not related to your PR

Comment (xiaoyu-work) [reply olive/workflows/run/config.py:69]: New `set_pass_flows` still needs a pass to be registered. But empty dict works after the change for `set_pass_flows`

Comment (trajepl) [reply olive/workflows/run/config.py:69]: Oh, xiaoyu is right. In my previous changes, I missed this part.
image

**Review by trajepl — APPROVED** [review]
